---
title: "Code to replicate plots in psupertime manuscript"
author: "Will Macnair"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_vignette:
    fig_caption: yes
vignette: >
  %\VignetteIndexEntry{Code to replicate plots in psupertime manuscript}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
library('psupertime')
```

# `psupplementary` overview

`psupplementary` is a supplementary R package for the package `psupertime`, which analyses single cell RNA-seq ("scRNAseq") data where groups of the cells have labels with a known or expected ordering (for example, samples from a time series experiment *day1*, *day2*, ..., *day5*). It uses *ordinal logistic regression* to identify a set of genes which recapitulate the group-level ordering for individual cells. The main `psupertime` package is available [here](github.com/wmacnair/psupertime).

This supplementary package contains both the data used for the analyses in the manuscript, and scripts required to replicate them. Single cell RNA-seq datasets are typically large, so to keep the main package light we created this package to contain the additional datasets.

(Our intention is to allow replication of all figures in the manuscript. At present the code allows replication of all analysis relating to Figure 1, including multiple supplementary figures; replication of the remaining figures will follow!)

## Datasets included

There are six scRNAseq datasets included in this package:

- *acinar_sce*, consisting of aging acinar cells, with sequential labels corresponding to age of donor in years, stored in the variable 'donor_age' (from [here](https://www.sciencedirect.com/science/article/pii/S009286741731053X), GSE81547)
- *germ_sce*, consisting of developing human female germline cells, with sequential labels corresponding to age in weeks, stored in the variable 'time' (from [here](https://www.cell.com/cell-stem-cell/fulltext/S1934-5909(17)30078-4), GSE86146)
- *beta_sce*, consisting of developing beta cells, with sequential labels corresponding to developmental stage, stored in the variable 'age' (from [here](https://www.cell.com/cell-metabolism/fulltext/S1550-4131(17)30208-5), GSE87375)
- *hesc_sce*, consisting of human embryonic stem cells, with sequential labels corresponding to embryonic day, stored in the variable 'esc_day' (from [here](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3929/), E-MTAB-3929)
- *mef_sce*, consisting of MEFs reprogrammed to neurons, with sequential labels corresponding to days since induction, stored in the variable 'time_point' (from [here](https://www.nature.com/articles/nature18323), GSE67310)
- *colon_sce*, consisting of human colon cells, where the sequential labels are derived from unsupervised clustering (from [here](https://www.cell.com/cell-systems/fulltext/S2405-4712(17)30449-0), GSE102698)

## Analysis of acinar cells

Here we illustrate `psupertime` on a simple example, using the default parameters to analyse pancreas cells taken from donors of varying ages.

```{r}
# NOT RUN
# acinar_cells_plots()
```

## GO enrichment of genes relevant to acinar cells

We can also analyse which GO terms are enriched at various points of the sequence.

```{r}
# NOT RUN
# acinar_cells_go_term_analysis()
```

## Exploratory data analysis of unlabelled data

Here we illustrate how `psupertime` can be combined with dimensionality reduction and unsupervised clustering to explore user-selected sequences of clusters. Here, we use UMAP for the dimensionality reduction, and Seurat for the unsupervised clustering; users may choose their preferred combination.

```{r}
# NOT RUN
# unsupervised_clustering_of_colon_cells()
```
